src/ape/utils/abi.py

```diff
@@ -111,6 +111,10 @@ def parse(self, output_types: List[ABIType], values: Union[List, Tuple]) -> Any:
                         }
                         item_type = ABIType.parse_obj(item_type_data)
                         parsed_item = self.parse([item_type], [value])
+                        # If it's an empty array of structs, replace `None` with empty list
+                        if output_type.type == "tuple[]" and parsed_item is None:
```

Comment:
What about fixed size, like `tuple[2]`?

Reply:
A lot more permutations to test, some examples:
```vyper
struct Foo:
    a: address
    b: bytes32
    c: uint256

...
def bar1() -> Foo:
...
def bar2() -> DynArray[Foo, N]:
...
def bar3() -> (DynArray[Foo, N], DynArray[Foo, N]):
...
def bar4() -> (Foo[N], DynArray[Foo, N]):
...
def bar5() -> DynArray[Foo[M], N]:
```

Reply:
I am adding these now and adjusting

Reply:
I added a bunch more tests cases that I think align with these.
Let me know if you think anything is missing yet!